api - adding a /stops endpoint

prhod opened the pull request:
Fixing https://github.com/Jungle-Bus/VapourTrail/issues/53
First step to remove JSON hack in t-rex tiles :  https://github.com/Jungle-Bus/VapourTrail/blob/master/imposm/post-process-display.sql#L335

frodrigo commented on api/api.py:
```diff
@@ -119,11 +119,58 @@ def get_route_stops_with_connections(route_id):
+    def get(self, stop_id):
+        if stop_id:
+            stops = get_stop(stop_id)
+            if len(stops) > 0 : 
```
Space at the end of line
```suggestion
            if len(stops) > 0 :
```

frodrigo commented on api/api.py:
```diff
@@ -119,11 +119,58 @@ def get_route_stops_with_connections(route_id):
+                has_tactile_paving,
+                has_departures_board,
+                is_wheelchair_ok,
+                st_asGeoJSON(ST_Transform(geom, 4326)) as geojson, 
```
```suggestion
                st_asGeoJSON(ST_Transform(geom, 4326)) as geojson,
```

frodrigo commented on api/api.py:
```diff
@@ -119,11 +119,58 @@ def get_route_stops_with_connections(route_id):
+        for k in d:
+            s[k] = d[k]
+        stops.append(s)
+    return stops
```
Why return a list of stops? and not only one.

prhod commented:
thanks for the review. I changed the list part returing only the first stop, and returning a 404 error if not found (instead of a 500).
I also changed "vapour_api" in "api" in dockerFile for local dev and docker consistancy.

nlehuby commented on api/api.py:
```diff
@@ -120,11 +120,54 @@ def get_route_stops_with_connections(route_id):
+                has_departures_board,
+                is_wheelchair_ok,
+                st_asGeoJSON(ST_Transform(geom, 4326)) as geojson,
+                routes_at_stop
```
the `routes_at_stop` contains many information that are not needed anymore.

This can be cleaned in another PR, but I think we need an issue to remember to do it ;)

nlehuby commented on api/api.py:
```diff
@@ -120,11 +120,54 @@ def get_route_stops_with_connections(route_id):
         return {"links": [{"href": "api.py/route/<route_id>"}]}
 
 
+osm_types = {0: "node", 1: "way", 2: "relation"}
```
There is an inconsistency in the endpoints:
In the `routes/route_osm_id` response, we get `stop_list[].osm_id` with the following format: `osm_type/osm_id` (for example "node/5257992450")

So we should expect the `stop_osm_id` to keep that format in the `stops` endpoint too (and we expect this to work `http://localhost:5000/stops/node/5257992450`)

But I think we actually need to fix the `routes` endpoint (so it's quite unrelated to these changes and it can be done in another PR)

frodrigo replied:
It's a bad idea in rest api to use "/" in id. I suggest to change the stop is format and use standard identifier chars in it.

prhod replied:
thanks for the comments. I've changed the URLs : 
- http://localhost:5000/stops/5257205236 : will assume the id is a node
- http://localhost:5000/stops/5257205236?osm_type=way : will search for a way

prhod commented:
I've modified the API: 
* the /stops/stop_id uses a node as default
* the /stops/osm_type/stop_id uses proper osm types

the response now look like: 
```json
{
    "has_departures_board": false,
    "has_shelter": false,
    "routes_at_stop": [
        {
            "origin": "Gare SNCF",
            "destination": "Parc d'activit\u00e9",
            "transport_mode": "bus",
            "name": "AggloBus - ligne B",
            "colour": "#AF2080",
            "route_osm_id": 7770652,
            "ref": "B",
            "network": "Agglobus",
            "operator": "Communaut\u00e9 d'Agglom\u00e9ration du Grand Gu\u00e9ret"
        }
    ],
    "local_ref": "",
    "name": "Labetoulle",
    "is_wheelchair_ok": false,
    "has_tactile_paving": false,
    "geojson": {
        "coordinates": [
            1.88312442602256,
            46.185857716226
        ],
        "type": "Point"
    },
    "has_bench": false,
    "osm_id": "node/5257205236"
}
```

The API response is modified at runtime, it may be done later in DB when fixing https://github.com/Jungle-Bus/VapourTrail/issues/56

prhod commented:
new proposal for response content using a full geosjon  : 
```json
{
    "type": "Feature",
    "geometry": {
        "type": "Point",
        "coordinates": [
            1.88312442602256,
            46.185857716226
        ]
    },
    "properties": {
        "has_tactile_paving": false,
        "osm_type": "node",
        "name": "Labetoulle",
        "has_bench": false,
        "routes_at_stop": [
            {
                "transport_mode": "bus",
                "destination": "Parc d'activit\u00e9",
                "operator": "Communaut\u00e9 d'Agglom\u00e9ration du Grand Gu\u00e9ret",
                "name": "AggloBus - ligne B",
                "route_osm_id": 7770652,
                "colour": "#AF2080",
                "origin": "Gare SNCF",
                "ref": "B",
                "network": "Agglobus"
            }
        ],
        "osm_id": 5257205236,
        "is_wheelchair_ok": false,
        "local_ref": "",
        "has_shelter": false,
        "has_departures_board": false
    }
}
```
